editor/changeset_wrapper.cpp

```diff
@@ -44,60 +50,47 @@ void ChangesetWrapper::LoadXmlFromOSM(ms::LatLon const & ll, pugi::xml_document
     MYTHROW(OsmXmlParseException, ("Can't parse OSM server response for GetXmlFeaturesAtLatLon request", response.second));
 }
 
-XMLFeature ChangesetWrapper::GetMatchingFeatureFromOSM(XMLFeature const & ourPatch, FeatureType const & feature)
+XMLFeature ChangesetWrapper::GetMatchingNodeFeatureFromOSM(m2::PointD const & center)
+{
+  // Match with OSM node.
+  ms::LatLon const ll = MercatorBounds::ToLatLon(center);
+  pugi::xml_document doc;
+  // Throws!
+  LoadXmlFromOSM(ll, doc);
+
+  pugi::xml_node const bestNode = GetBestOsmNode(doc, ll);
+  if (bestNode.empty())
+    MYTHROW(OsmObjectWasDeletedException,
+            ("OSM does not have any nodes at the coordinates", ll, ", server has returned:", doc));
+
+  return XMLFeature(bestNode);
+}
+
+XMLFeature ChangesetWrapper::GetMatchingAreaFeatureFromOSM(set<m2::PointD> const & geometry)
```

Comment:
I do not see where the std::set methods except begin() and end() used here. How about to pass an std::vector?

Generally speaking, if you need to represent a set of elements, and you do not need to dynamically add/remove elements from it, std::vector + SortUnique() is the best choice.